sample/src/main/kotlin/com/slack/circuit/sample/home/HomePresenter.kt

```diff
@@ -116,33 +145,89 @@ class HomeUiFactory @Inject constructor() : Ui.Factory {
-        title = {
-          Text("Adoptables", fontSize = 22.sp, color = MaterialTheme.colorScheme.onPrimaryContainer)
-        },
-        colors =
-          TopAppBarDefaults.centerAlignedTopAppBarColors(
-            containerColor = MaterialTheme.colorScheme.primaryContainer
+  val modalState =
+    rememberModalBottomSheetState(
+      initialValue =
+        if (state.petListFilterState.showBottomSheet) ModalBottomSheetValue.Expanded
+        else ModalBottomSheetValue.Hidden
+    )
+
+  // Monitor bottom sheet state and emit event whenever the user dismisses the modal
+  LaunchedEffect(modalState) {
+    snapshotFlow { modalState.isVisible }
+      .collect { isVisible ->
+        // Toggle if state says the modal should be visible but the snapshot says it isn't.
+        if (state.petListFilterState.showBottomSheet && !isVisible)
+          eventSink(
+            HomeScreen.Event.PetListFilterEvent(PetListFilterScreen.Event.ToggleAnimalFilter)
           )
-      )
-    },
-    bottomBar = {
-      BottomNavigationBar(selectedIndex = state.homeNavState.index) { index ->
-        eventSink(HomeScreen.Event.HomeEvent(HomeNavScreen.Event.HomeNavEvent(index)))
+      }
+  }
+
+  ModalBottomSheetLayout(
+    sheetState = modalState,
+    sheetContent = {
+      Column {
+        GenderFilterOption(state.petListFilterState, eventSink)
+        SizeFilterOption(state.petListFilterState, eventSink)
       }
     }
   ) {
-    if (state.homeNavState.index == DOGS_SCREEN_INDEX) {
-      PetList(modifier = Modifier.padding(it), state = state.petListState) { event ->
-        eventSink(HomeScreen.Event.PetListEvent(event))
+    Scaffold(
+      modifier = Modifier.navigationBarsPadding().systemBarsPadding().fillMaxWidth(),
+      topBar = {
+        CenterAlignedTopAppBar(
+          title = {
+            Text(
+              "Adoptables",
+              fontSize = 22.sp,
+              color = MaterialTheme.colorScheme.onPrimaryContainer
+            )
+          },
+          colors =
+            TopAppBarDefaults.centerAlignedTopAppBarColors(
+              containerColor = MaterialTheme.colorScheme.primaryContainer
+            ),
+          actions = {
+            IconButton(
+              onClick = {
+                eventSink(
+                  HomeScreen.Event.PetListFilterEvent(PetListFilterScreen.Event.ToggleAnimalFilter)
+                )
+              }
+            ) {
+              androidx.compose.material.Icon(
+                imageVector = Icons.Default.FilterList,
+                contentDescription = "filter pet list",
+                tint = Color.White
+              )
+            }
+          },
+        )
+      },
+      bottomBar = {
+        BottomNavigationBar(selectedIndex = state.homeNavState.index) { index ->
+          eventSink(HomeScreen.Event.HomeEvent(HomeNavScreen.Event.HomeNavEvent(index)))
+        }
+      }
+    ) { paddingValues ->
+      if (state.homeNavState.index == DOGS_SCREEN_INDEX) {
```

Comment:
nit: can we abstract this now and hide the screen detail from this UI?

```kotlin
val screen = homeScreens.get(index, state.petListFilterState.gender, state.petListFilterState.size)
CircuitContent(screen, onNavEvent = ...)
```

PetListScreen would use the extra values, About would ignore them

Reply:
Addressed here, https://github.com/slackhq/circuit/pull/101/commits/89de7f6d29b498d957c17c4ec559be364ecd09c1.